### PR TITLE
Streaks & Progress, Simplified "up next" card.

### PR DESCRIPTION
**Pull request by bradmontgomery**
This PR replaces the 2nd card in the feed with a progress / streaks card. The streaks card shows the past 7 days of user activity, and a bar chart with the number of actions performed.

The previous 'up next' card has been simplified, as well.

### CODE REVIEW

**Comment by Revenaunt** [comment]
👍 

I believe the ProgressCircleView entry in build.gradle can be removed, but I gotta do some maintenance in the file anyway, so it's ok if you don't want to do that.

**Comment by bradmontgomery** [comment]
@Revenaunt I will probably let you clean up the `build.gradle` file, and ditch the ProgressCircleView stuff.  :-P

File: src/main/java/org/tndata/android/compass/ui/StreakView.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        mStreak = streak;
+        init();
+    }
+
+    public void init() {
+        if(mStreak != null) {
+            setOrientation(LinearLayout.VERTICAL);
+            setGravity(Gravity.CENTER_VERTICAL);
+
+            LayoutInflater inflater = (LayoutInflater) mContext.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
```
Comment (Revenaunt) [inline src/main/java/org/tndata/android/compass/ui/StreakView.java]: LayoutInflater.from(getContext()); <- Shortcut ;)

File: src/main/java/org/tndata/android/compass/adapter/feed/MainFeedAdapter.java [hunk 2]
```diff
@@ -225,6 +233,10 @@ public void onBindViewHolder(RecyclerView.ViewHolder rawHolder, int position){
         else if (CardTypes.isUpNext(position)){
             ((UpNextHolder)rawHolder).bind(mFeedData.getUpNextAction(), mFeedData.getProgress());
         }
+        //Streaks
+        else if (CardTypes.isStreaks(position)){
+            ((StreaksHolder)rawHolder).bind(mFeedData.getStreaks());
```
Comment (Revenaunt) [inline src/main/java/org/tndata/android/compass/adapter/feed/MainFeedAdapter.java]: You bind every time the dataset is updated (for all intents and purposes)

File: src/main/java/org/tndata/android/compass/ui/StreakView.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+            inflater.inflate(R.layout.streak, this, true);
```
Comment (Revenaunt) [inline src/main/java/org/tndata/android/compass/ui/StreakView.java]: `true` adds the newly inflated view to `this`, which is the current `StreakView`

File: src/main/java/org/tndata/android/compass/ui/StreakView.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    private TextView mStreakDay;
+
+    public StreakView(Context context, AttributeSet attrs) {
+        super(context, attrs);
+        mContext = context;
+    }
+
+    public void setStreak(FeedData.Streak streak) {
+        mStreak = streak;
+        init();
```
Comment (Revenaunt) [inline src/main/java/org/tndata/android/compass/ui/StreakView.java]: Call to init: https://github.com/tndatacommons/android-app/pull/215/files#r71026004

File: src/main/java/org/tndata/android/compass/adapter/feed/StreaksHolder.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    void bind(List<FeedData.Streak> streaks){
+        mStreaks = streaks;
+        if(streaks.size() == 7) {
+            day0.setStreak(streaks.get(0));
+            day1.setStreak(streaks.get(1));
+            day2.setStreak(streaks.get(2));
+            day3.setStreak(streaks.get(3));
+            day4.setStreak(streaks.get(4));
+            day5.setStreak(streaks.get(5));
+            day6.setStreak(streaks.get(6));
```
Comment (Revenaunt) [inline src/main/java/org/tndata/android/compass/adapter/feed/StreaksHolder.java]: 7 calls to setStreak: https://github.com/tndatacommons/android-app/pull/215/files#r71026054

File: src/main/java/org/tndata/android/compass/adapter/feed/MainFeedAdapter.java (hunk 2 above)
Comment (Revenaunt) [inline src/main/java/org/tndata/android/compass/adapter/feed/MainFeedAdapter.java]: This bind method gets called every time the data-set is updated (for all intents and purposes). https://github.com/tndatacommons/android-app/pull/215/files#r71026170

**Comment by bradmontgomery** [comment]
@Revenaunt take a look at the last 2 commits on this, and see if those look OK. This appears to fix the issue with multiple progress bars, and this appears to work on my phone.

**Comment by Revenaunt** [comment]
Looks good to me. :+1:

**Pull request merged**